htdocs/core/modules/accountancy/mod_bookkeeping_argon.php

```diff
@@ -0,0 +1,159 @@
+	public $error = '';
+
+	/**
+	 * @var string name
+	 */
+	public $name = 'Argon';
+
+
+	/**
+	 * Constructor
+	 */
+	public function __construct()
+	{
+		global $conf, $mysoc;
+	}
+
+	/**
+	 *  Return description of numbering module
+	 *
+	 *	@param	Translate	$langs      Lang object to use for output
+	 *  @return string      			Descriptive text
+	 */
+	public function info($langs): string
+	{
+		global $langs;
+		return $langs->trans("BookkeepingNumRefModelDesc");
+	}
+
+
+	/**
+	 *  Return an example of numbering
+	 *
+	 *  @return     string      Example
+	 */
+	public function getExample(): string
+	{
+		return "2025VT0001";
+	}
+
+
+	/**
+	 *  Checks if the numbers already in the database do not
+	 *  cause conflicts that would prevent this numbering working.
+	 *
+	 *  @param  BookKeeping	$object		Object we need next value for
+	 *  @return boolean     				false if conflict, true if ok
+	 */
+	public function canBeActivated($object): bool
+	{
+		global $conf, $langs, $db;
+
+		$max = '';
+
+		$prefix = $this->getPrefix($object);
+		// If prefix smaller than 6, prefix is not correct (should be 6 or 7, YYYYCC OR YYYYCCC)
+		if (!empty($prefix) || strlen($prefix) < 6) {
+			$langs->load("errors");
+			$this->error = $langs->trans('ErrorNumRefModel', $max);
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * 	Return next free value
+	 *
+	 *  @param  BookKeeping	$object		Object we need next value for
+	 *  @return string|int<-1,0>		Value if OK, -1 if KO
+	 */
+	public function getNextValue(BookKeeping $object)
+	{
+		global $db, $conf;
+
+		$prefix = $this->getPrefix($object);
+		$posindice = strlen($prefix) + 1;
+		$sql = "SELECT MAX(CAST(SUBSTRING(ref FROM ".$posindice.") AS SIGNED)) as max";
+		$sql .= " FROM {$db->prefix()}accounting_bookkeeping";
+		$sql .= " WHERE ref LIKE '{$db->escape($prefix)}%'";
+		$sql .= " AND entity = ".getEntity($object->element);
+
+
+		$resql = $db->query($sql);
+		if ($resql) {
+			$obj = $db->fetch_object($resql);
+			if ($obj) {
+				$max = intval($obj->max);
+			} else {
+				$max = 0;
+			}
+		} else {
+			dol_syslog("mod_bookkeeping_argon::getNextValue", LOG_DEBUG);
+			return -1;
+		}
+		if ($max >= (pow(10, 4) - 1)) {
+			$num = $max + 1; // If counter > 9999, we do not format on 4 chars, we take number as it is
+		} else {
+			$num = sprintf("%04d", $max + 1);
+		}
+
+		dol_syslog("mod_bookkeeping_argon::getNextValue return {$prefix}{$num}");
+		return "{$prefix}{$num}";
+	}
+
+	/**
+	 * Returns the prefix for current Bookkeeping object
+	 * Year used in prefix is the beginning fiscal year.
+	 *
+	 * @param BookKeeping $object	Book keeping record
+	 * @return string
+	 */
+	private function getPrefix(BookKeeping $object): string
+	{
+		$fiscalStartMonth = getDolGlobalInt('SOCIETE_FISCAL_MONTH_START') ?: 1;
+		$docYear = dol_print_date($object->doc_date, '%Y');
+		$docMonth = dol_print_date($object->doc_date, '%m');
+		$docFiscalYear = $docMonth < $fiscalStartMonth ? $docYear - 1 : $docYear;
```

Comment:
Use parenthesis here. Result is suspicious without.

Reply:
On line 153, it's better with : $fiscalStartMonth = getDolGlobalInt('SOCIETE_FISCAL_MONTH_START', 1);